airflow/jobs/scheduler_job.py

```diff
@@ -629,13 +629,18 @@ def _process_task_instances(self, dag, task_instances_list, session=None):
         active_dag_runs = []
         for run in dag_runs:
             self.log.info("Examining DAG run %s", run)
-            # don't consider runs that are executed in the future
+            # don't consider runs that are executed in the future unless
+            # specified by config and schedule_interval is None
             if run.execution_date > timezone.utcnow():
-                self.log.error(
-                    "Execution date is in future: %s",
-                    run.execution_date
-                )
-                continue
+                if run.dag.schedule_interval or not conf.getboolean(
```

Comment:
This can now be:

```python
            if run.execution_date > timezone.utcnow() and not run.dag.allow_future_exec_dates:
                    self.log.error(
                        "Execution date is in future: %s",
                        run.execution_date
                    )
                    continue
```